lib/utils/KalturaCache.js

```diff
@@ -132,58 +134,60 @@ KalturaCache = {
-		var lockKey = 'Lock' + entryRequired;
-		this.add(lockKey, 1, KalturaConfig.config.cache.entryRequired, function() { // add fails if key exists
-			KalturaLogger.debug("Removing " + renditionId + " from " + entryRequired + " values");
+		var lockKey = this.getKey(this.LOCK_KEY_PREFIX, [entryRequired]);
+		var lockTimeout = 50;
+		if (KalturaConfig.config.cache.lock) {
+			lockTimeout = KalturaConfig.config.cache.lock;
+		}
+		var recall = function() {
+			setTimeout(function () {
+				This.removeRenditionIdFromEntryRequiredValue(entryRequired, renditionId, retries++);
+			}, lockTimeout * 2 * Math.random());
+		};
 
+		this.add(lockKey, 1, lockTimeout, function() { // add fails if key exists
+			KalturaLogger.debug("Removing " + renditionId + " from " + entryRequired + " values");
 			This.get(entryRequired, function (data) {
-				var currentRenditions = This.extractEntryRequiredValue(data);
-				var newEntryRequired = "\n";
-				var shouldChange = false;
-				var foundValidRendition = false;
-				for (var i = 0; i < currentRenditions.length; i++) {
-					if (currentRenditions[i].indexOf(renditionId) == -1) {
-						if (currentRenditions[i].trim().length !== 0) {
-							foundValidRendition = true;
-						}
-						newEntryRequired = newEntryRequired.concat("\n").concat(currentRenditions[i].trim());
-					} else {
-						shouldChange = true;
-					}
-				}
-				if (shouldChange) {
-					if (foundValidRendition) {
+				var newEntryRequired;
+				var index = data.indexOf(renditionId);
+				if ( index != -1) {
+					newEntryRequired = data.substr(0, index).concat(data.substr(index + renditionId.length));
+					if (newEntryRequired.trim().length > 0) {
 						This.set(entryRequired, newEntryRequired, KalturaConfig.config.cache.entryRequired,
 							function () {
 								This.del(lockKey);
 								KalturaLogger.debug("Successfully removed rendition " + renditionId + " from entryRequired " + entryRequired +
-								" new value is : " + newEntryRequired);
+									" new value is : " + newEntryRequired);
 							},
 							function () {
 								This.del(lockKey);
 								KalturaLogger.debug("Failed to remove rendition " + renditionId + " from entryRequired " + entryRequired +
-								" ,Though should have - retrying");
-								setTimeout(function () {
-									This.removeRenditionIdFromEntryRequiredValue(entryRequired, renditionId);
-								}, 100);
+									" ,Though should have - retrying");
+								recall();
 							});
 					} else {
+						KalturaLogger.debug("After deletion of rendition " + renditionId + " entryRequired: " + entryRequired + " is empty - deleting it");
 						This.del(entryRequired);
 						This.del(lockKey);
 					}
-
 				} else {
+					// this means that the rendition was not part of the required entry moving on
 					This.del(lockKey);
 				}
 			}, function () {
 				This.del(lockKey);
+				KalturaLogger.debug("Failed to get " + entryRequired + " key ,Though should have - retrying");
+				recall();
```

Comment:
If failed , no need to recall, and make sure that KalturaError is printed since this case is an absurd.